java/code/src/com/redhat/rhn/frontend/xmlrpc/systemgroup/ServerGroupHandler.java

```diff
@@ -545,4 +556,112 @@ private ServerGroup lookup(Integer id, User user) {
             throw new TaskomaticApiException(e.getMessage());
         }
     }
+
+    /**
+     * List all Configuration Channels assigned to a system group
+     * @param loggedInUser the user
+     * @param systemGroupName the group name
+     * @return list of Config Channels
+     * @since 25
+     *
+     * @xmlrpc.doc List all Configuration Channels assigned to a system group
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("string", "systemGroupName")
+     * @xmlrpc.returntype
+     * #array_begin()
+     * $ConfigChannelSerializer
+     * #array_end()
+     */
+    public List<ConfigChannel> listAssignedConfigChannels(User loggedInUser, String systemGroupName) {
+        ServerGroup group = Optional.ofNullable(
+                ServerGroupFactory.lookupByNameAndOrg(systemGroupName, loggedInUser.getOrg())
+                ).orElseThrow(() -> new LookupServerGroupException(systemGroupName));
+
+        return StateFactory.latestConfigChannels(group).orElseGet(Collections::emptyList);
+    }
+
+    /**
+     * Subscribe given config channels to a system group
+     * @param loggedInUser the user
+     * @param systemGroupName the group name
+     * @param configChannelLabels list of config channel labels to subscribe
+     * @return 1 on success
+     *
+     * @xmlrpc.doc Subscribe given config channels to a system group
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("string", "systemGroupName")
+     * @xmlrpc.param #array_single("string", "configChannelLabels")
+     * @xmlrpc.returntype 1 on success, exception on failure
+     */
+    public int subscribeConfigChannel(User loggedInUser, String systemGroupName,
+            List<String> configChannelLabels) {
+
+        ServerGroup group = Optional.ofNullable(
+                ServerGroupFactory.lookupByNameAndOrg(systemGroupName, loggedInUser.getOrg())
+                ).orElseThrow(() -> new LookupServerGroupException(systemGroupName));
+        ConfigurationManager manager = ConfigurationManager.getInstance();
+
+        List<ConfigChannel> channels = new LinkedList<>();
+        for (String l: configChannelLabels) {
```

Comment:
you could replace this for loop with stream.map here and in below

Reply:
done